src/services/refactors/addOrRemoveBracesToArrowFunction.ts

```diff
@@ -15,33 +15,61 @@ namespace ts.refactor.addOrRemoveBracesToArrowFunction {
         addBraces: boolean;
     }
 
+    type InfoOrError = {
+        info: Info,
+        error?: never
+    } | {
+        info?: never,
+        error: string
+    };
+
     function getAvailableActions(context: RefactorContext): readonly ApplicableRefactorInfo[] {
         const { file, startPosition, triggerReason } = context;
         const info = getConvertibleArrowFunctionAtPosition(file, startPosition, triggerReason === "invoked");
         if (!info) return emptyArray;
 
-        return [{
-            name: refactorName,
-            description: refactorDescription,
-            actions: [
-                info.addBraces ?
-                    {
-                        name: addBracesActionName,
-                        description: addBracesActionDescription
-                    } : {
-                        name: removeBracesActionName,
-                        description: removeBracesActionDescription
-                    }
-            ]
-        }];
+        if (info.error === undefined) {
+            return [{
+                name: refactorName,
+                description: refactorDescription,
+                actions: [
+                    info.info.addBraces ?
+                        {
+                            name: addBracesActionName,
+                            description: addBracesActionDescription
+                        } : {
+                            name: removeBracesActionName,
+                            description: removeBracesActionDescription
+                        }
+                ]
+            }];
+        }
+
+        if (context.preferences.provideRefactorNotApplicableReason) {
+            return [{
+                name: refactorName,
+                description: refactorDescription,
+                actions: [{
+                    name: addBracesActionName,
+                    description: addBracesActionDescription,
+                    notApplicableReason: info.error

```

Comment:
Without having read the code in detail, it seems intuitively unlikely to me that add-braces and remove-braces would fail for the same reason.